ocp_resources/resource.py

```diff
@@ -76,6 +83,29 @@ def _get_api_version(dyn_client: DynamicClient, api_group: str, kind: str) -> st
     return res.group_version
 
 
+@contextlib.contextmanager
+def get_retry_cluster_exceptions_client(
+    client: DynamicClient,
+    backoff_factor: float = 0.5,
+    retry_count: int = 5,
+    status_codes=None,
+) -> Any:
+    original_retries = client.configuration.retries
+    if not status_codes:
+        status_codes = set()
+    client.configuration.retries = urllib3.util.Retry(
+        total=retry_count,
+        backoff_factor=backoff_factor,
+        allowed_methods=(urllib3.util.Retry.DEFAULT_ALLOWED_METHODS | {http.HTTPMethod.PATCH, http.HTTPMethod.POST}),
+        status_forcelist=(DEFAULT_HTTP_STATUS_CODE_TO_RETRY | status_codes),
+        raise_on_redirect=False,
+        raise_on_status=False,
```

Comment:
Note that it might change the exposed "API" of the openshift-python-wrapper. It will not raise MaxRetryError anymore, instead it will raise the underlying exception (usually a kubernetes.ApiException). IMHO, this is a better behaviour but it might not be backward compatible.